Compilation error in svg parser

### LVGL version

v9.3.0

### Platform

ESP-IDF 5.4.1
Visual Studio Code, ESP-IDF plugin 1.10.0
Hardware ESP32S3 Dev module

### What happened?

With SVG option enabled,, a number of compile errors appear

### How to reproduce?

For example: 
`In function '_skip_space_and_separators':   
C:/Projects/button30/managed_components/lvgl__lvgl/src/libs/svg/lv_svg_parser.c:438:39: error: array subscript has type 'cha 
' [-Werror=char-subscripts]
  438 |     while((str < str_end) && (isspace(*str) || _is_separators(*str))) {`

I was able to fix this by casting the `*str` to `(unsigned char)` in the call to `isspace`, but I'm sure there are better solutions.

Steps to reproduce:

- Use ESP-IDF extension (version 1.10.2) in Visual Studio Code, in my case on Windows 11
- Install ESP-IDF  v5.4.2 (under Advanced section in extension) if not already installed
- Create a New Project with ESP-IDF V5.4.2 ->  Target IDF_TARGET esp32S3, choose hello_world template
- Open Terminal Shell and add LVGL like this: ```idf.py add-dependency "lvgl/lvgl^9.3.0"```
- Add ```REQUIRES lvgl``` to the CMakelist.txt file of the hello_world project
- Now ```idf.py menuconfig``` (maybe build project first, not sure if needed)
- In menuconfig change ```LV_VECTOR_GRAPHICS``` to yes
- In menuconfig change ```LV_USE_SVG```
- Save the configuration and start the build

Boom, next  errors occur:
```
In file included from C:/Projects/lvgl542/managed_components/lvgl__lvgl/src/libs/svg/lv_svg_parser.c:13:
C:/Projects/lvgl542/managed_components/lvgl__lvgl/src/libs/svg/lv_svg_parser.c: In function '_skip_space':
C:/Projects/lvgl542/managed_components/lvgl__lvgl/src/libs/svg/lv_svg_parser.c:425:38: error: array subscript has type 'char' [-Werror=char-subscripts]
  425 |     while((str < str_end) && isspace(*str)) {
      |                                      ^~~~
C:/Projects/lvgl542/managed_components/lvgl__lvgl/src/libs/svg/lv_svg_parser.c: In function '_skip_space_and_separators':  
C:/Projects/lvgl542/managed_components/lvgl__lvgl/src/libs/svg/lv_svg_parser.c:438:39: error: array subscript has type 'char' [-Werror=char-subscripts]
  438 |     while((str < str_end) && (isspace(*str) || _is_separators(*str))) {
      |                                       ^~~~
C:/Projects/lvgl542/managed_components/lvgl__lvgl/src/libs/svg/lv_svg_parser.c: In function '_parse_color':
C:/Projects/lvgl542/managed_components/lvgl__lvgl/src/libs/svg/lv_svg_parser.c:518:28: error: array subscript has type 'char' [-Werror=char-subscripts]
  518 |             if(isxdigit(str[1]) && isxdigit(str[2]) && isxdigit(str[3])) {
      |                         ~~~^~~
C:/Projects/lvgl542/managed_components/lvgl__lvgl/src/libs/svg/lv_svg_parser.c:518:48: error: array subscript has type 'char' [-Werror=char-subscripts]
  518 |             if(isxdigit(str[1]) && isxdigit(str[2]) && isxdigit(str[3])) {
      |                                             ~~~^~~
C:/Projects/lvgl542/managed_components/lvgl__lvgl/src/libs/svg/lv_svg_parser.c:518:68: error: array subscript has type 'char' [-Werror=char-subscripts]
  518 |             if(isxdigit(str[1]) && isxdigit(str[2]) && isxdigit(str[3])) {
      |                                                                 ~~~^~~
C:/Projects/lvgl542/managed_components/lvgl__lvgl/src/libs/svg/lv_svg_parser.c:529:28: error: array subscript has type 'char' [-Werror=char-subscripts]
  529 |             if(isxdigit(str[1]) && isxdigit(str[2]) && isxdigit(str[3])
      |                         ~~~^~~
C:/Projects/lvgl542/managed_components/lvgl__lvgl/src/libs/svg/lv_svg_parser.c:529:48: error: array subscript has type 'char' [-Werror=char-subscripts]
  529 |             if(isxdigit(str[1]) && isxdigit(str[2]) && isxdigit(str[3])
      |                                             ~~~^~~```

etc...

As stated, easy to fix with a cast, but these are not my files.

--- a/src/libs/svg/lv_svg_parser.c
+++ b/src/libs/svg/lv_svg_parser.c
@@ -422,7 +422,7 @@ static bool _is_number_begin(char ch)
 
 static const char * _skip_space(const char * str, const char * str_end)
 {
-    while((str < str_end) && isspace(*str)) {
+    while((str < str_end) && isspace((unsigned char) * str)) {
         ++str;
     }
     return str;
@@ -435,7 +435,7 @@ static bool _is_separators(char c)
 
 static const char * _skip_space_and_separators(const char * str, const char * str_end)
 {
-    while((str < str_end) && (isspace(*str) || _is_separators(*str))) {
+    while((str < str_end) && (isspace((unsigned char) * str) || _is_separators(*str))) {
         ++str;
     }
     return str;
@@ -515,7 +515,7 @@ static const char * _parse_color(const char * str, const char * str_end, uint32_
 
     if(*str == '#') {
         if(len == 4) { // three digit hex format '#rgb'
-            if(isxdigit(str[1]) && isxdigit(str[2]) && isxdigit(str[3])) {
+            if(isxdigit((unsigned char)str[1]) && isxdigit((unsigned char)str[2]) && isxdigit((unsigned char)str[3])) {
                 char st[3] = {0};
                 st[0] = st[1] = str[1];
                 r = (uint8_t)strtol(st, NULL, 16);
@@ -526,8 +526,8 @@ static const char * _parse_color(const char * str, const char * str_end, uint32_
             }
         }
         else if(len == 7) {    // six digit hex format '#rrggbb'
-            if(isxdigit(str[1]) && isxdigit(str[2]) && isxdigit(str[3])
-               && isxdigit(str[4]) && isxdigit(str[5]) && isxdigit(str[6])) {
+            if(isxdigit((unsigned char)str[1]) && isxdigit((unsigned char)str[2]) && isxdigit((unsigned char)str[3])
+               && isxdigit((unsigned char)str[4]) && isxdigit((unsigned char)str[5]) && isxdigit((unsigned char)str[6])) {
                 char st[3] = {0};
                 st[0] = str[1];
                 st[1] = str[2];
@@ -1462,7 +1462,7 @@ static void _process_paint(lv_svg_node_t * node, lv_svg_attr_type_t type, const 
             url_start = ptr + 1;
         }
 
-        while((ptr < val_end) && !isspace(*ptr) && *ptr != ')') {
+        while((ptr < val_end) && !isspace((unsigned char) * ptr) && *ptr != ')') {
             ++ptr;
         }
 

--- a/src/libs/thorvg/tvgInitializer.cpp
+++ b/src/libs/thorvg/tvgInitializer.cpp
@@ -74,7 +74,7 @@ static bool _buildVersionInfo(uint32_t* major, uint32_t* minor, uint32_t* micro)
     uint32_t microVal = atoi(p);
 
     char sum[7];
-    snprintf(sum, sizeof(sum), "%d%02d%02d", majorVal, minorVal, microVal);
+    snprintf(sum, sizeof(sum), "%" PRIu32 "%02" PRIu32 "%02" PRIu32, majorVal, minorVal, microVal);
     _version = atoi(sum);
 
     if (major) *major = majorVal;

